core/domain/summary_services.py

```diff
@@ -163,6 +164,70 @@ def get_displayable_collection_summary_dicts_matching_ids(collection_ids):
     return _get_displayable_collection_summary_dicts(collection_summaries)
 
 
+def get_exp_metadata_dicts_matching_query(query_string, search_cursor, user_id):
+    """Given a query string and a search cursor, returns a list of exploration
+    dicts that satisfy the search query.
+
+    Args:
+        query_string: str. The search query for which the search is to be
+            performed.
+        search_cursor: str or None. The cursor location to start the search
+            from. If None, the returned values are from the beginning
+            of the results list.
+        user_id: str. The user id of editor.
```

Comment:
I don't understand what's going on here, from reading this docstring. How does the user_id affect what's returned by this function?